common/src/main/java/org/terracotta/angela/common/tcconfig/holders/TcConfigHolder.java

```diff
@@ -128,7 +129,7 @@ public void initialize(PortAllocator portAllocator, Function<String, Boolean> su
         Node hostNode = (Node) xPath.evaluate("@host", server, XPathConstants.NODE);
         String hostname =
             hostNode == null || hostNode.getTextContent().equals("%i") || hostNode.getTextContent()
-                .equals("%h") ? "localhost" : hostNode.getTextContent();
+                .equals("%h") ? IpUtils.getHostName() : hostNode.getTextContent();
```

Comment:
this is wrong to default to `localhost`. Correct core behavior is to put the hostname for %h